rs/tests/networking/canister_http_correctness_test.rs

```diff
@@ -610,30 +814,1006 @@ pub fn test_http_calls_to_ic_fails(env: TestEnv) {
+    assert_http_response(&response);
+}
+
+fn test_post_call(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+    let expected_body = "POST";
+
+    let url = format!("https://[{}]:20443/{}", webserver_ipv6, "anything");
+    let body = Some("hello_world".as_bytes().to_vec());
+    let headers = BoundedHttpHeaders::new(vec![
+        HttpHeader {
+            name: "name1".to_string(),
+            value: "value1".to_string(),
+        },
+        HttpHeader {
+            name: "name2".to_string(),
+            value: "value2".to_string(),
+        },
+    ]);
+    let max_response_bytes = Some(666);
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers,
+        method: HttpMethod::POST,
+        body,
+        transform: Some(TransformContext {
+            function: TransformFunc(candid::Func {
+                principal: get_proxy_canister_id(&env).into(),
+                method: "transform".to_string(),
+            }),
+            context: vec![0, 1, 2],
+        }),
+        max_response_bytes,
+    };
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: request.clone(),
+            cycles: 500_000_000_000,
+        },
+    ))
+    .expect("Request succeeds.");
+
+    assert_matches!(&response, RemoteHttpResponse {body, status: 200, ..} if body.contains(expected_body));
+    assert_distinct_headers(&response);
+    assert_http_json_response(&request, &response);
+}
+
+/// Send 6666 repeating `x` to /anything endpoint.
+/// Use HEAD http method. It only asks for the head, not the body.
+/// Set max response size to 666 (order of magnitude smaller)
+fn test_head_call(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+
+    let long_x_string = "x".repeat(6666);
+    let url = format!(
+        "https://[{}]:20443/{}/{}",
+        webserver_ipv6, "anything", long_x_string
+    );
+    let body = Some("hello_world".as_bytes().to_vec());
+    let headers = BoundedHttpHeaders::new(vec![
+        HttpHeader {
+            name: "name1".to_string(),
+            value: "value1".to_string(),
+        },
+        HttpHeader {
+            name: "name2".to_string(),
+            value: "value2".to_string(),
+        },
+    ]);
+    let max_response_bytes = Some(666);
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers,
+        method: HttpMethod::HEAD,
+        body,
+        transform: None,
+        max_response_bytes,
+    };
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: request.clone(),
+            cycles: 500_000_000_000,
+        },
+    ))
+    .expect("Request succeeds.");
+
+    assert_matches!(&response, RemoteHttpResponse { status: 200, .. });
+    assert_distinct_headers(&response);
```

Comment:
let's also assert that the returned content-length is equal to 6666 (the use case of making a head request)